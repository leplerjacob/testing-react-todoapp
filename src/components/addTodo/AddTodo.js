import { useState } from "react";
import "./addtodo.css";

export default function AddTodo(props) {
  const [todo, setTodo] = useState("");
  const [date, setDate] = useState("");

  const addTodo = props.addTodo;

  function changeValue(e) {
    switch (e.target.name) {
      case "todo": {
        setTodo(e.target.value);
      }
      case "date": {
        setDate(e.target.value);
      }
      default: {
        return;
      }
    }
  }

  function createTodoAndSend() {
    const createdTodo = {
      id: null,
      todo: todo,
      date: date,
    };
    addTodo(createdTodo);
  }

  return (
    <form>
      <label htmlFor="todo">To Do</label>
      <input
        name="todo"
        type="text"
        id="input todo"
        value={todo}
        onChange={changeValue}
      />
      <label htmlFor="date">Date</label>
      <input
        name="date"
        type="date"
        id="input date"
        value={date}
        onChange={changeValue}
      />
      <label htmlFor="submit">Submit</label>
      <input type="button" value="Add Todo" onClick={createTodoAndSend} />
    </form>
  );
}
