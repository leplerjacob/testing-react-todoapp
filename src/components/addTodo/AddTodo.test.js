import React from "react";
import {render} from '@testing-library/react'
import renderer from "react-test-renderer";
import AddTodo from "./AddTodo";

describe("Todo properties", () => {
  let component;
  beforeEach(() => {
    component = render(<AddTodo />);
  });

  // 'yarn jest -u' updates snapshot
  test("Renders todo form component", () => {
    const tree = renderer.create(<AddTodo />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Default date on form is current date", () => {
      
      // handles offset timezone
      let dateNow = new Date();
      const offset = dateNow.getTimezoneOffset();
      dateNow = new Date(dateNow.getTime() - (offset * 60 - 1000))
      .toISOString()
      .split("T")[0];
      
      // const dateInput = component.container.querySelector('#date')
      const dateInput = component.queryByPlaceholderText(`${dateNow}`)

    //   expect(dateInput)
  });
});

// test("Updates value of input on change", () => {});

// test("Form resets input values on submit", () => {});

/* 3 methods to get text rendered on component: 
    - .toHaveTextContent to search element for text
    - .getByText to retrieve element by text
    - .querySelector then expect .toHaveTextContent
*/

/* Print render component with .debug()
 */
