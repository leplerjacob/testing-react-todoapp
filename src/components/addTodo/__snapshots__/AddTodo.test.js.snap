// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo properties Renders todo form component 1`] = `
<form>
  <label
    htmlFor="todo"
  >
    To Do
  </label>
  <input
    id="todo"
    name="todo"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    htmlFor="date"
  >
    Date
  </label>
  <input
    id="date"
    name="date"
    onChange={[Function]}
    placeholder="YYYY-MM-DD"
    type="date"
    value=""
  />
  <label
    htmlFor="submit"
  >
    Submit
  </label>
  <input
    id="submit"
    onClick={[Function]}
    type="button"
    value="Add Todo"
  />
</form>
`;
